tardis/model.py

```diff
@@ -268,6 +268,16 @@ def to_hdf(self, path_or_buf, path='', plasma_properties=None):
         None
 
         """
+        properties = ['t_inner', 'ws', 't_rads', 'v_inner', 'v_outer']
+        for p in properties:
+            output_value = getattr(self, p)
+            if hasattr(output_value, 'si'):
+                output_value = output_value.si.value
+            if np.isscalar(output_value):
+                output_value = [output_value]
+            data = pd.DataFrame(output_value)
```

Comment:
You want to do it this way:
```python
df = pd.DataFrame()
df['a'] = [1,2,3]
df['b'] = [1,2]
```

Fotis Tsamis <notifications@github.com> schrieb am Mo., 20. Juni 2016 um
18:56 Uhr:

> In tardis/model.py
> <https://github.com/tardis-sn/tardis/pull/589#discussion_r67725293>:
>
> > @@ -268,6 +268,16 @@ def to_hdf(self, path_or_buf, path='', plasma_properties=None):
> >          None
> >
> >          """
> > +        properties = ['t_inner', 'ws', 't_rads', 'v_inner', 'v_outer']
> > +        for p in properties:
> > +            output_value = getattr(self, p)
> > +            if hasattr(output_value, 'si'):
> > +                output_value = output_value.si.value
> > +            if np.isscalar(output_value):
> > +                output_value = [output_value]
> > +            data = pd.DataFrame(output_value)
>
> As far as I checked it does not do that automatically.
>
> >>>pd.DataFrame({'a': [1,2], 'b' : [1,2,3]})ValueError: arrays must all be same length
>
> while:
>
> >>> pd.DataFrame({'a': [1,2], 'b' : [1,2]})
>    a  b0  1  11  2  2
>
> So one would have to take care of manually adding np.nan for the missing
> values of shorter columns as well.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/tardis-sn/tardis/pull/589/files/b9ca672ee6796a356ca4593f1e8886adfb82397a..d8f3a53a09090c6946f8f366ae5e376fb6e68b4c#r67725293>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF4OjnmBdZ7eP1VmVYuyElrjD-I0Rjd_ks5qNsZEgaJpZM4I1pWa>
> .
>